lib/pam/pam.go

```diff
@@ -311,11 +338,45 @@ func (p *PAM) Close() error {
 
 	// Release strings that were allocated when opening the PAM context.
 	C.free(unsafe.Pointer(p.service_name))
-	C.free(unsafe.Pointer(p.user))
+	C.free(unsafe.Pointer(p.login))
+	C.free(unsafe.Pointer(p.loginContext))
```

Comment:
If `Close` returns an error, these `free` calls never run and (presumably) the values leak.  Is it possible to `defer` these at the beginning of `Close`, or is it unsafe to `free` these unless all of the pam close/end stuff was successful?